Adding support for local, custom ADCIRC src.

wwlwpd opened the pull request:
Issue-748: This will allow users with ADCIRC source somewhere on their
ASGS host machine to build and register ADCIRC via `build adcirc`, by
selecting the "custom" option and answering the questions in the interactive
mode that follows. Upon successful compilation, this ADCIRC is available
like all others via "load adcirc" and can be seen along with the others
via "list adcirc". Patches are not supported at this time.

Resolves #748.

wwlwpd commented:
@jasonfleming this needs to be tested _and_reviewed. @notstarboard should also take a look at it. Let me know when y'all can discuss and I can describe the testing procedure; really just requires local file access to the original "source" (the original source is left untouched, just `cp`'d).

jasonfleming commented:
Will work on this on hatteras; need a copy of the source code ; copy to work dir where asgs is setup ; start shell ; run build adcirc ; last menu option is custom ; (11 for now) ; will ask for path to source ; tld  not work ; ask for name etc ; skips patching and git checkout ; copies source from you dir to normal asgs place ; from there identical to previous procedure ; adds to adcirc meta dir file with info needed ; also produces json file ; new key called `custom_src` plus env var ; if it is pointing to custom source this has full path ; otherwise 0 value ; once built successfully works just like any other adcirc build ; `load adcirc x` etc ; will get path to source via email

wwlwpd commented:
rebased on current master SHA, f3f89d82687802d88f79a301213abc73c19aa9b2

jasonfleming commented:
Thank you @wwlwpd I will be testing this today and providing feedback.

wwlwpd commented:
@dave-gso has reported an interest issue, I will try to reproduce and fix if found.

jasonfleming commented on bin/init-adcirc.sh:
```diff
@@ -190,38 +239,56 @@ ADCSWAN_MAKE_CMD="make $ADCSWAN_BINS compiler=${ADCIRC_COMPILER} MACHINENAME=${A
+  cp -rf $CUSTOM_SRC/* $ADCIRCBASE
+  # clean up destination
+  echo " ... cleaning up cruft from '$ADCIRCBASE'"
+  rm -rf $ADCIRCBASE/.git                2> /dev/null
```
suggest to use `make clean` or similar

wwlwpd replied:
good idea, will do!

jasonfleming reviewed (dismissed):
This looks great @wwlwpd thank you. Suggest additional comments to document any assumptions or caveats, e.g., that the `make` build method is used (not cmake), and that the netcdf paths should not be hard coded in the developers' build system.

wwlwpd commented:
@jasonfleming I will commit to completing #975, to document `build adcirc` in the wiki once this is merged - just to keep the scope of the work down.

wwlwpd commented:
Latest commit,

```
commit 708c8328a3bff0573e5c125b0359246f1264074c
Author: wwlwpd <46434714+wwlwpd@users.noreply.github.com>
Date:   Fri Aug 12 04:09:28 2022 -0400

    fixup! Adding support for custom paths to ADCIRC src.
    
    Replaced series of `rm -rv` with `make clobber`; detects if
    target directory exists, and if so will offer to delete it. The
    default for this is set to N (no). Added fun Star Wars ASCII art
    that is shown whenever ADCIRC build succeeds.
```

wwlwpd commented:
Preliminary documentatio, https://github.com/StormSurgeLive/asgs/wiki/ASGS-Compiling-ADCIRC

jasonfleming reviewed (approved):
Hey @wwlwpd I tested this on hatteras, and was able to build @dave-gso weir_pump code successfully. I executed `adcirc` on the command line in the asgs shell and it was able to start successfully without any library errors. I was then able to build v55.01 (bonus testing) successfully and test run `adcirc` on the command line. Looks great!

wwlwpd commented:
Thanks for testing, will squash and merge!

wwlwpd commented:
Merged manually. Approved by @jasonfleming
